src/main/commands/scan_async.cc

```diff
@@ -66,8 +69,19 @@ NAN_METHOD(AerospikeClient::ScanAsync)
 		as_v8_debug(log, "Using scan ID %lli for async scan.", scan_id);
 	}
 
-	as_v8_debug(log, "Sending async scan command");
-	status = aerospike_scan_async(client->as, &cmd->err, p_policy, &scan, &scan_id, async_scan_listener, cmd, NULL);
+	as_partition_filter_set_all(&pf);
+	if (partitions_from_jsobject(&pf, info[2].As<Object>(), log) == AS_NODE_PARAM_OK) {
```

Comment:
What if the partitions object is invalid? Shouldn't the client abort the whole operation in that case with an error, rather than just silently ignore it?